hudi-cli/src/main/java/org/apache/hudi/cli/commands/LockAuditingCommand.java

```diff
@@ -185,4 +236,323 @@ public String showLockAuditStatus() {
+      Option<Long> endTimestamp = Option.empty();
+      for (int i = jsonObjects.size() - 1; i >= 0; i--) {
+        JsonNode obj = jsonObjects.get(i);
+        if (obj.has("state") && "END".equals(obj.get("state").asText())) {
+          endTimestamp = Option.of(obj.has("timestamp") ? obj.get("timestamp").asLong() : 0L);
+          break;
+        }
+      }
+      
+      // Find last expiration time as fallback
+      Option<Long> lastExpirationTime = Option.empty();
+      if (!jsonObjects.isEmpty()) {
+        JsonNode lastObject = jsonObjects.get(jsonObjects.size() - 1);
+        if (lastObject.has("lockExpiration")) {
+          lastExpirationTime = Option.of(lastObject.get("lockExpiration").asLong());
+        }
+      }
+      
+      return Option.of(new TransactionWindow(
+          ownerId,
+          transactionStartTime,
+          startTimestamp,
+          endTimestamp,
+          lastExpirationTime,
+          filename
+      ));
+    } catch (Exception e) {
+      LOG.warn("Failed to parse audit file: " + filename, e);
+      return Option.empty();
+    }
+  }
+
+  /**
+   * Validates transaction windows for overlaps and proper closure.
+   */
+  private ValidationResults validateTransactionWindows(List<TransactionWindow> windows) {
+    List<String> errors = new ArrayList<>();
+    List<String> warnings = new ArrayList<>();
+    
+    // Check for transactions without proper END
+    for (TransactionWindow window : windows) {
+      if (!window.endTimestamp.isPresent()) {
+        warnings.add(String.format("[WARNING] %s => transaction did not end gracefully. This could be due to driver OOM or non-graceful shutdown.", 
+            window.filename));
+      }
+    }
+    
+    // Sort windows by start time for overlap detection
+    List<TransactionWindow> sortedWindows = new ArrayList<>(windows);
+    sortedWindows.sort(Comparator.comparingLong(w -> w.startTimestamp));
+    
+    // Check for overlaps
+    for (int i = 0; i < sortedWindows.size(); i++) {
+      TransactionWindow currentWindow = sortedWindows.get(i);
+      long currentEnd = currentWindow.getEffectiveEndTime();
+      
+      // Check all subsequent windows for overlaps
+      for (int j = i + 1; j < sortedWindows.size(); j++) {
+        TransactionWindow otherWindow = sortedWindows.get(j);
+        long otherStart = otherWindow.startTimestamp;
+        
+        // Check if windows overlap
+        if (otherStart < currentEnd) {
+          // There's an overlap - check if current transaction ended properly before the other started
+          if (currentWindow.endTimestamp.isPresent() && currentWindow.endTimestamp.get() <= otherStart) {
+            // Transaction ended properly before the other started - no issue
+          } else {
+            // Either no END timestamp or END timestamp is after other transaction started
+            errors.add(String.format("[ERROR] %s => overlaps with %s", 
+                currentWindow.filename, otherWindow.filename));
```

Comment:
same: these shld be PoJos so they are extensible in future. Not sure if we added Lombok in hudi..